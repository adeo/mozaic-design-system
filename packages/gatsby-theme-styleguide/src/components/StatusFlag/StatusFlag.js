import * as React from 'react'
import { checkStatus } from '../../utils/status'
import * as styles from '../PatternStatusGroup/status.module.css'

const StatusFlag = ({ status }) => {
  const platformStatus = checkStatus(status)

  if (!status || !platformStatus) return false

  return (
    <span
      className={`${styles.statusFlag} ${styles[platformStatus]}`}
      status={platformStatus}
    >
      {platformStatus}
    </span>
  )
}

export default StatusFlag
