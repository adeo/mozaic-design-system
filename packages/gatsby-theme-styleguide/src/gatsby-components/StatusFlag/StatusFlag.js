import React from 'react'
import styled from 'styled-components'
import { checkStatus } from '../../utils/status'

const Flag = styled.span`
  background: #fff;
  border: solid 1px transparent;
  border-radius: 2px;
  font-size: 0.875rem;
  line-height: 1;
  padding: 0.2rem 0.5rem 0.3rem;

  ${({ status }) => {
    return status === 'stable'
      ? `
        border-color: #58e3b7;
        color: #328269;
    `
      : status === 'wip'
      ? `
        border-color: #e3cb58;
        color: #a58e1b;
    `
      : /*bÃªta*/ `
        border-color: #e35876;
        color: #82325e;
    `
  }}
`

const StatusFlag = ({ status }) => {
  if (!status) return false
  const authorizedStatus = checkStatus(status)
  if (!authorizedStatus) return false
  return <Flag status={authorizedStatus}>{authorizedStatus}</Flag>
}

export default StatusFlag
