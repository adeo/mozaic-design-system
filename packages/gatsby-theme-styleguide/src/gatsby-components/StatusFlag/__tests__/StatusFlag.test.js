import React from 'react'
import renderer from 'react-test-renderer'
import StatusFlag from '../StatusFlag'
import 'jest-styled-components'

describe(`<StatusFlag>`, () => {
  it(`should render correctly with a wip status`, () => {
    const tree = renderer.create(<StatusFlag status="wip" />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border', 'solid 1px #e3cb58')
  })

  it(`should render correctly with a bêta status`, () => {
    const tree = renderer.create(<StatusFlag status="bêta" />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border', 'solid 1px #e35876')
  })

  it(`should render correctly with a stable status`, () => {
    const tree = renderer.create(<StatusFlag status="stable" />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border', 'solid 1px #58e3b7')
  })

  it(`should should not render anything using an unsuported status prop`, () => {
    const tree = renderer.create(<StatusFlag status="foo" />).toJSON()
    expect(tree).toBe(null)
  })
})
