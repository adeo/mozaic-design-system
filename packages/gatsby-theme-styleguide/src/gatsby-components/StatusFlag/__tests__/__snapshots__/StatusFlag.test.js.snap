// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusFlag> should render correctly with a bêta status 1`] = `
.c0 {
  background: #fff;
  border: solid 1px transparent;
  border-radius: 2px;
  font-size: 0.875rem;
  line-height: 1;
  padding: 0.2rem 0.5rem 0.3rem;
  border-color: #e35876;
  color: #82325e;
}

<span
  className="c0"
>
  bêta
</span>
`;

exports[`<StatusFlag> should render correctly with a stable status 1`] = `
.c0 {
  background: #fff;
  border: solid 1px transparent;
  border-radius: 2px;
  font-size: 0.875rem;
  line-height: 1;
  padding: 0.2rem 0.5rem 0.3rem;
  border-color: #58e3b7;
  color: #328269;
}

<span
  className="c0"
>
  stable
</span>
`;

exports[`<StatusFlag> should render correctly with a wip status 1`] = `
.c0 {
  background: #fff;
  border: solid 1px transparent;
  border-radius: 2px;
  font-size: 0.875rem;
  line-height: 1;
  padding: 0.2rem 0.5rem 0.3rem;
  border-color: #e3cb58;
  color: #a58e1b;
}

<span
  className="c0"
>
  wip
</span>
`;
